timer.js

```diff
@@ -25,48 +20,170 @@ function startTimer(durationSeconds = 25 * 60) {
     state.timer.remainingTimeOnResume = timerDurationS;
-
+    console.log(state.timer.remainingTimeOnResume);
+    timerDurationS -= 1;
+    // decrement the timer by 1s
+    currentTomatoStartEnd();
     // handle when the timer runs out
     if (timerDurationS < 0) {
       clearInterval(state.timerIntervalId);
       if (state.isWorkInterval) {
         startTimer(300); // Start rest interval
         state.isWorkInterval = false;
+        updateCurrentTomato();
       } else {
-        updateTomatoCount();
         startTimer(1500); // Start work interval
         state.isWorkInterval = true;
       }
     }
   }, 1 * 1000);
-}
+  } else if (!state.isWorkInterval) {
+
+    let minutes = 0;
+    let seconds = 0;
+    
+    // activate the timer CSS
+    const timerDisplay = document.getElementById("timer");
+    timerDisplay.classList.add("timer-active"); // Add class when timer starts
+    
+    //   start the timer...
+    state.timerIntervalId = setInterval(function updateTimerCountdown() {
+      minutes = parseInt(timerDurationS / 60, 10); // 25
+      seconds = parseInt(timerDurationS % 60, 10); // 0
+  
+      minutes = minutes < 10 ? "0" + minutes : minutes;
+      seconds = seconds < 10 ? "0" + seconds : seconds;
+  
+      // updating the UI with the state
+      document.getElementById("timerMinutes").textContent = minutes;
+      document.getElementById("timerSeconds").textContent = seconds;
+      document.getElementById("timer").classList.add("timer-active");
+      // save the current remaining time to state
+      state.timer.remainingTimeOnResume = timerDurationS;
+    
+      timerDurationS -= 1;
+      // handle when the timer runs out
+      if (timerDurationS < 0) {
+        clearInterval(state.timerIntervalId);
+        if (state.isWorkInterval) {
+          startTimer(300); // Start rest interval
+          state.isWorkInterval = false;       
+        } else {
+          startTimer(1500); // Start work interval
+          state.isWorkInterval = true;
+        }
+      }
+    }, 1 * 1000);
+  } else {
+    alert("Chose todo from table");
+    return;
+  }
+  
+} 
 
 function pauseTimer() {
   clearInterval(state.timerIntervalId);
   document.getElementById("timer").classList.remove("timer-active"); // Remove class when paused
+  updateCurrentTomato();
 }
 
 function stopTimer() {
-  //updateTomatoCount();
   clearInterval(state.timerIntervalId);
+  state.isWorkInterval = !state.isWorkInterval; // Toggle to next interval
+  state.timer.remainingTimeOnResume = undefined; // Reset remaining time
+  state.currentId = undefined;
+  state.currentTodo = undefined;
+
+  if (state.isWorkInterval) {
+    timerDurationS = 25 * 60;
+  } else {
+    timerDurationS = 5 * 60;
+  }
+
+  minutes = parseInt(timerDurationS / 60, 10); // 25
+  seconds = parseInt(timerDurationS % 60, 10); // 0
+
+  minutes = minutes < 10 ? "0" + minutes : minutes;
+  seconds = seconds < 10 ? "0" + seconds : seconds;
+
+  // updating the UI with the state
```

Comment:
I notice this pattern repeating "updating the UI with the state" in a couple of places, could we clean up by abstracting that pattern into a function like this?

`function updateUI(state){ // handle updating UI with state in here }`

Reply:
I made this func:
![image](https://github.com/antondurinin/ToDo/assets/167963627/ccd1523d-96e6-42c1-813b-26dcbf8f09b3)